go/loopback/main.go

```diff
@@ -32,49 +37,72 @@ func Init(homeDir string, logFile string, runModeStr string, serverURI string, a
 			fmt.Println("Error decoding run mode", err, runModeStr)
 		}
 		config := libkb.AppConfig{HomeDir: homeDir, LogFile: logFile, RunMode: runMode, Debug: true, LocalRPCDebug: "Acsvip", ServerURI: serverURI, SecurityAccessGroupOverride: accessGroupOverride}
-		err = libkb.G.Configure(config, usage)
+		err = kbCtx.Configure(config, usage)
 		if err != nil {
 			panic(err)
 		}
 
-		service := (service.NewService(g, false))
-		service.StartLoopbackServer()
-		service.G().SetService()
+		svc := service.NewService(kbCtx, false)
+		svc.StartLoopbackServer()
+		kbCtx.SetService()
+		kbCtx.SetUIRouter(service.NewUIRouter(kbCtx))
 
 		serviceLog := config.GetLogFile()
 		logs := libkb.Logs{
 			Service: serviceLog,
 		}
 
 		logSendContext = libkb.LogSendContext{
-			Contextified: libkb.NewContextified(service.G()),
+			Contextified: libkb.NewContextified(kbCtx),
 			Logs:         logs,
 		}
 
+		// Uncomment after we KBFS is ready
+		// kbfsParams := libkbfs.DefaultInitParams(kbCtx)
```

Comment:
Hmm...I don't think this'll work. There's no circular dependency in the modules, but there will be a circular dependency in the repos!

I think you'll need an executable that lives in the kbfs repo (or even a third repo) that starts up both the loopback server and libkbfs and hooks them up. That means vendoring more stuff into kbfs, but that seems like the least bad option.

I'm guessing for mobile everything has to live in a single executable? I.e., you can't have a keybase daemon and the kbfs daemon as separate executables talking to each other?